libs/core/libs/shared/src/api.rs

```diff
@@ -719,3 +752,67 @@ impl Request for AdminRebuildIndexRequest {
     const METHOD: Method = Method::POST;
     const ROUTE: &'static str = "/admin-rebuild-index";
 }
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum StripeAccountState {
+    Ok,
+    InvoiceFailed,
+    Canceled,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum GooglePlayAccountState {
+    Ok,
+    Canceled,
+    GracePeriod,
+    OnHold,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum AppStoreAccountState {
+    Ok,
+    GracePeriod,
+    FailedToRenew,
+    Expired,
+}
+
+impl FromStr for StripeAccountState {
+    type Err = ();
```

Comment:
I wonder if this is gonna be a nightmare to debug, I think even `String` could be fine with an error message, maybe you can include `s` in the error.